src/v/kafka/server/handlers/describe_groups.cc

```diff
@@ -46,22 +35,17 @@ ss::future<response_ptr> describe_groups_handler::handle(
     request.decode(ctx.reader(), ctx.header().version);
     klog.trace("Handling request {}", request);
 
-    return ss::do_with(
-      describe_groups_ctx(std::move(ctx), std::move(request), ssg),
-      [](describe_groups_ctx& octx) {
-          return ss::parallel_for_each(
-                   octx.request.data.groups.begin(),
-                   octx.request.data.groups.end(),
-                   [&octx](kafka::group_id id) {
-                       return octx.rctx.groups()
-                         .describe_group(std::move(id))
-                         .then([&octx](described_group g) {
-                             octx.response.data.groups.push_back(std::move(g));
-                         });
-                   })
-            .then(
-              [&octx] { return octx.rctx.respond(std::move(octx.response)); });
-      });
+    std::vector<ss::future<described_group>> described;
+
+    for (auto& group_id : request.data.groups) {
```

Comment:
wow that's a lot of cleanup! 

(cc: @mmaslankaprv i wonder how raft will start to look as we start to untangle some of the futures for coros)